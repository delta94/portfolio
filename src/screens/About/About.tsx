import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Element } from 'react-scroll';
import { AboutInfo, Avatar, ThreeDAnimation, VanishAnimation } from '../../components';
import { CustomTheme } from '../../theme';
import { MemoizedComponent } from '../../utils/custom-hook';

interface AboutProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  header: {
    color: theme.colors.white,
    marginLeft: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center',
    cursor: 'pointer',
  },
  avatar: {
    width: '50%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  aboutInfo: {
    width: '50%',
    paddingRight: theme.spacing(3.75),
  },
  aboutInfoMobile: {
    width: '90%',
    paddingRight: theme.spacing(0),
    marginTop: theme.spacing(2),
  },
  aboutContainer: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
  },
  aboutContainerMobile: {
    flexWrap: 'wrap',
  },
}));

const About: React.FC<AboutProps> = () => {
  const classes: Record<
    'header' | 'avatar' | 'aboutInfo' | 'aboutContainer' | 'aboutContainerMobile' | 'aboutInfoMobile',
    string
  > = useStyles();
  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });

  return (
    <Element name='About'>
      <div className={classes.header}>{MemoizedComponent(<VanishAnimation words={['About', 'Me']} />)}</div>
      <div className={clsx([classes.aboutContainer, isMobile && classes.aboutContainerMobile])}>
        <div className={classes.avatar}>
          {MemoizedComponent(
            <ThreeDAnimation>
              <Avatar url='https://lh3.googleusercontent.com/pw/ACtC-3fkT2wC_oB8Y4Fo48qOW6acvyLmAhr7NoFlLzb1Vnh-5qY_FDc_1ktlCP42JrwK5_TpT6PyAJZ18v63iAk-hIGQ6xnQEDw1agvzeGw_U8j-RN6ipnl6Z-1bHyfdL8jbLAXmXPJRNaGN1Qi6oq0k-6ntEQ=w800-h741-no' />
            </ThreeDAnimation>,
          )}
        </div>
        <div className={clsx([classes.aboutInfo, isMobile && classes.aboutInfoMobile])}>
          {MemoizedComponent(<AboutInfo />)}
        </div>
      </div>
    </Element>
  );
};

export { About };
