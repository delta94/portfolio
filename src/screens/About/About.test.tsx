import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { About } from './About';

test('About renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <About />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
