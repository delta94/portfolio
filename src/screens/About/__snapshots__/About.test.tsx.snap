// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About renders correctly 1`] = `
<DocumentFragment>
  <div
    name="About"
  >
    <div
      class="header-0-2-1"
    >
      <div
        style="opacity: 0; transform: translate3d(0,20px,0);"
      >
        <h2
          class="h2Text-0-2-7"
        >
          About
        </h2>
      </div>
      <div
        style="opacity: 0; transform: translate3d(0,20px,0);"
      >
        <h2
          class="h2Text-0-2-7"
        >
          Me
        </h2>
      </div>
    </div>
    <div
      class="aboutContainer-0-2-5"
    >
      <div
        class="avatar-0-2-2"
      >
        <div
          style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
        >
          <img
            alt="Avatar"
            class="mainContainer-0-2-8"
          />
        </div>
      </div>
      <div
        class="aboutInfo-0-2-3"
      >
        <div
          class="mainContainer-0-2-9"
        >
          <div
            class="triangleContainer-0-2-18"
          >
            <div
              class="triangle-0-2-19"
            />
          </div>
          <div
            class="descriptionParentContainer-0-2-22"
          >
            <div
              class="descriptionContainer-0-2-10"
            >
              <div
                class="description-0-2-12"
              >
                I am Shaikh Zeeshan Murshed, Front-End developer from Montreal, Canada. I have experience in making Website. Also I am a hobbyist photographer.
              </div>
              <div
                class="button-0-2-14"
              >
                <button
                  class="mainContainer-0-2-24"
                  style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
                >
                  Download My CV
                </button>
              </div>
            </div>
            <div
              class="skillsInfoContainer-0-2-16"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
